src/components/search-results/search-results.tsx

```diff
@@ -0,0 +1,57 @@
+import { useContext } from 'react';
+import { NavLink, useParams } from 'react-router-dom';
+import { appContext } from '../../App-context';
+import { getItems, searchItems } from '../../requests/requests';
+import s from './search-results.module.css';
+
+const SearchResults = () => {
+  const { page } = useParams();
+  const context = useContext(appContext);
+  if (context!.isLoading) {
+    return <div className={s.loader}></div>;
+  }
+
+  const sendDetaitsRequest = async (pageNumber: number, index: number) => {
+    const requestValue = localStorage.getItem('Input value') || '';
+    context!.setIsDetailsLoading(true);
```

Comment:
`    const requestValue = localStorage.getItem('Input value') || '';

    context.setIsDetailsLoading(true);

    const searchResponse = requestValue
      ? await searchItems(requestValue, pageNumber)
      : await getItems(pageNumber);

    context.setIsDetailsLoading(false);

    if (searchResponse) {`

Reply:
It is preferable to split logic with empty lines. When using the conditions, the positive case should go first.